Userland/Applications/ImageViewer/ViewWidget.cpp

```diff
@@ -246,6 +246,9 @@ ErrorOr<void> ViewWidget::try_open_file(String const& path, Core::File& file)
         frames.ensure_capacity(decoded_image->frames.size());
         for (u32 i = 0; i < decoded_image->frames.size(); i++) {
             auto& frame_data = decoded_image->frames[i];
+            if (!frame_data.bitmap) {
```

Comment:
Why do we need that?
A failure to decode should be caught by the `!decoded_image.has_value()` above IMO

Reply:
I've changed the `Client::decode_image` call so that it returns empty if any of the frames it receives are null. 

All of the users of this function were either doing the null check themselves anyway, or just ignoring the possibility of null frames, so doing it this way simplifies things.